homeassistant/components/openweathermap/sensor.py

```diff
@@ -161,16 +163,24 @@ async def async_setup_entry(
     name = domain_data.name
     weather_coordinator = domain_data.coordinator
 
-    entities: list[AbstractOpenWeatherMapSensor] = [
-        OpenWeatherMapSensor(
-            name,
-            f"{config_entry.unique_id}-{description.key}",
-            description,
-            weather_coordinator,
+    if domain_data.mode == OWM_MODE_FREE_FORECAST:
+        entity_registry = er.async_get(hass)
+        entries = er.async_entries_for_config_entry(
+            entity_registry, config_entry.entry_id
         )
-        for description in WEATHER_SENSOR_TYPES
-    ]
-    async_add_entities(entities)
+        for entry in entries:
+            entity_registry.async_remove(entry.entity_id)
```

Comment:
So wait why do we do this?

Reply:
free mode 'forecast' has only forecast values, it doesn't have current weather values, so all these sensors are redundant with this mode